src/main/js/directives/application.html

```diff
@@ -12,7 +12,12 @@ <h2>{{application.haku.name }}</h2>
                target="_blank">{{ localization('button.showApplication') }}</a>
             <span ng-if="application.tuloskirje && application.tuloskirje.createdDate">
                 <span class="delimiter"> | </span>
-                <a class="tuloskirje" ng-href="{{url('omatsivut.applications.tuloskirje',application.haku.oid)}}" ng-show="application.hasForm" ng-class="{disabled: applicationForm.$dirty|| statusMessageType=='pending' || isValidating}" target="_blank">
+                <a class="tuloskirje" ng-if="!application.token" ng-href="{{url('omatsivut.applications.tuloskirje',application.haku.oid)}}" ng-show="application.hasForm" ng-class="{disabled: applicationForm.$dirty|| statusMessageType=='pending' || isValidating}" target="_blank">
```

Comment:
```suggestion
                <a class="tuloskirje" ng-href="{{ application.token ? url('omatsivut.applications.tuloskirje.token',application.token,application.haku.oid) : url('omatsivut.applications.tuloskirje',application.haku.oid) }}" ng-show="application.hasForm" ng-class="{disabled: applicationForm.$dirty|| statusMessageType=='pending' || isValidating}" target="_blank">
```

Tämän voisi varmaan esittää näin, niin ei tarvitsisi toistaa tuota samaa pätkää kahdesti kun vain url muuttuu.